### ISSUE TITLE
[Bug]: FocusZoneCallout autoscrolls on click of certain elements

### ISSUE THREAD

**Issue by amdeenmsft** [microsoft/fluentui#27288]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz
    Memory: 8.22 GB / 31.71 GB
  Browsers:
    Edge: Spartan (44.22621.1413.0), Chromium (111.0.1661.51)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://codepen.io/amnddeen/pen/mdGQVoM

## Actual Behavior
https://codepen.io/amnddeen/pen/mdGQVoM

FocusTrapCallout auto scrolls on click of callout body, FluentUI Checkbox, and FluentUI ChoiceGroup.

Repro steps:

- Add some kind of input to FocusTrapCallout
- Add enough elements to cause callout body to scroll
- Add checkbox/choice group field and text field to bottom of callout
- Scroll to bottom. On click of any part of the callout body, checkbox, or choice group, the callout will auto scroll to the top without changing selection of field. Notice that clicking the text field does NOT cause the callout to scroll

This is an A11y issue. When user is zoomed in far enough, the callout will render a scrollbar and these fields will be unusable.

Note:  unable to access (http://aka.ms/fluentui-a11y)

## Expected Behavior
User should be able to interact with callout elements regardless of scroll bar

### Requested priority

Blocking

### Products/sites affected

Microsoft Power Pages

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by behowell** [comment]
It looks like focus is getting moved back to the text field at the top of the callout on mouse-down, when clicking anywhere on the callout. Need to investigate why FocusTrapZone is doing this.

**Comment by spmonahan** [comment]
Configuring `focusTrapProps` will allow you to get the behavior you want:

```tsx
<FocusTrapCallout
  focusTrapProps={{
    forceFocusInsideTrap: false,
    isClickableOutsideFocusTrap: true
  }}
/>
```

I've coded up an example here: https://codepen.io/seanms/pen/dyqEmXG?editors=0010
